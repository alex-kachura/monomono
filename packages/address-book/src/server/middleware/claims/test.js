describe('Claims middleware', () => {
  let mockExpress;
  let mockReq;
  const mockConfig = {
    cookie: {
      userAccessToken: {
        name: 'accessToken',
      },
      UUID: {
        name: 'uuid',
      },
    },
  };

  beforeEach(() => {
    mockReq = {
      region: 'gb',
    };
    mockExpress = {
      request: mockReq,
    };

    jest.doMock('express', () => mockExpress);
    jest.doMock('config', () => mockConfig);

    require('./');
  });

  afterEach(() => {
    jest.resetModules();
    Reflect.deleteProperty(require.cache, require.resolve('./')); // eslint-disable-line
  });

  it('should add a "getClaims" method to the response object', () => {
    expect(mockReq).toHaveProperty('getClaims');
    expect(typeof mockReq.getClaims).toBe('function');
  });

  describe('getClaims', () => {
    it('should return an object with an access token', () => {
      const accessToken = 'test';

      mockReq.cookies = {
        accessToken,
      };

      expect(mockReq.getClaims()).toHaveProperty('accessToken', accessToken);
    });

    it('should return an object with a UUID', () => {
      const uuid = 'uuid';

      mockReq.cookies = {
        uuid,
      };

      expect(mockReq.getClaims()).toHaveProperty('uuid', uuid);
    });
  });
});
