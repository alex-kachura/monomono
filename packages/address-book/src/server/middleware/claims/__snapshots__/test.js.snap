// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Claims middleware getClaims should return an object with an access token and UUID 1`] = `
Object {
  "accessToken": "test",
  "uuid": "uuid",
}
`;
